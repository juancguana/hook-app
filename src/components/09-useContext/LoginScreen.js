import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);
  const user = {
    id: 123,
    name: 'Juan',
  };
  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <button
        className='btn btn-primary'
        onClick={() => setUser(user)}
      >Login</button>
    </div>
  );
};
