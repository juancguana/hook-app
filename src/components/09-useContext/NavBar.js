import React, { useState } from 'react';
import { Link, NavLink } from 'react-router-dom';

export const NavBar = () => {
  const [isNavCollapsed, setIsNavCollapsed] = useState(true);

  const handleNavCollapse = () => setIsNavCollapsed(!isNavCollapsed);
  return (
    <nav className='navbar navbar-expand-lg navbar-light bg-light'>
      <div className='container-fluid'>
        <Link  className='navbar-brand' to='/'>
          useContex
        </Link>
        <button
          className='navbar-toggler'
          type='button'
          data-bs-toggle='collapse'
          data-bs-target='#navbarNavAltMarkup'
          aria-controls='navbarNavAltMarkup'
          aria-expanded={!isNavCollapsed ? true : false}
          onClick={handleNavCollapse}
          aria-label='Toggle navigation'
        >
          <span className='navbar-toggler-icon'></span>
        </button>
        <div
          className={`${isNavCollapsed ? 'collapse' : ''} navbar-collapse`}
          id='navbarNav'
        >
          <ul className='navbar-nav'>
            <NavLink 
            end
              activeclassname='active'
              className='nav-link'
              aria-current='page'
              to='/'
            >
              Home
            </NavLink>
            <NavLink  activeclassname='active' className='nav-link' to='/about'>
              About
            </NavLink>
            <NavLink  activeclassname='active' className='nav-link' to='/login'>
              Login
            </NavLink>
          </ul>
        </div>
      </div>
    </nav>
  );
};
