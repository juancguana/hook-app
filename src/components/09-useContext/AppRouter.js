import React from 'react';
import {
  BrowserRouter as Router,
  Routes,
  Route,
  Navigate,
} from 'react-router-dom';
import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <div className='container'>
          <Routes>
            <Route exact path='/' element={<HomeScreen />} />
            <Route path='/about' element={<AboutScreen />} />
            <Route path='/login' element={<LoginScreen />} />
            <Route path='*' element={<Navigate to='/' />} />
          </Routes>
        </div>
      </div>
    </Router>
  );
};
