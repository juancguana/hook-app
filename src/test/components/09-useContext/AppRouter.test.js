import { mount } from 'enzyme';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Test in <AppRouter />', () => {
  const user = { id: 1, name: 'Juan' };
  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <AppRouter />
    </UserContext.Provider>
  );

  test('should show correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
