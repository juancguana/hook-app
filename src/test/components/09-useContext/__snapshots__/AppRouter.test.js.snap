// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <AppRouter /> should show correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        useContex
      </a>
      <button
        aria-controls="navbarNavAltMarkup"
        aria-expanded={false}
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarNavAltMarkup"
        data-bs-toggle="collapse"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          className="navbar-nav"
        >
          <a
            activeclassname="active"
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            activeclassname="active"
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            activeclassname="active"
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
   "id": 1,
   "name": "Juan"
}
      </pre>
    </div>
  </div>
</div>
`;
